staging/src/k8s.io/apiserver/pkg/endpoints/handlers/rest.go

```diff
@@ -318,3 +319,7 @@ func parseTimeout(str string) time.Duration {
 	}
 	return 30 * time.Second
 }
+
+func isDryRun(url *url.URL) bool {
+	return len(url.Query()["dry-run"]) != 0
```

Comment:
It's probably OK for this sort of PR (maybe @deads2k won't think so), but you should be aware that this skips all of our ordinary URL parsing code.

Reply:
> It's probably OK for this sort of PR (maybe @deads2k won't think so), but you should be aware that this skips all of our ordinary URL parsing code.

I do think it's an example of the feature branch allowing flexibility we wouldn't allow into master.  I think that flexibility is very useful when working something like this, but it does undermine the idea that the feature branch is as rigorous as master.

On a related note.  If dry-run works well in this branch and the semantics are well described, it seems like a good target to bring into master separately.

Reply:
This is targeting master :)

The goal is to get dry run as a concept established as early as possible so that it can be relied on to fail safe, since the consequences of submitting a dry run request the server doesn’t know is a dry run is really bad.

Reply:
Before the parameter is actually used for anything other than a top-level short-circuiting fail like this, I’d expect the create/update/patch handlers to be updated to use the *Options approach and decode parameters just like the other handlers.

Reply:
Yeah, this change is for master and wouldn't make sense for the feature branch.

I agree w/ @liggitt that it needs to go through the right stack before we do anything productive with it. I think this is fine as an easy & safe way to reserve it, though.

Reply:
looking at all the other `*Options` structs, we'd add this as `dryRun`, not `dry-run`